mailpoet/lib/Util/License/Features/CapabilitiesManager.php

```diff
@@ -115,4 +116,12 @@ public function getCapabilities(): Capabilities {
       $this->getSegmentFiltersLimit(),
     );
   }
+
+  public function showUpgradePage(): bool {
+    $tier = $this->getTier();
+    if ($this->subscribersFeature->hasValidPremiumKey() && isset($tier) && $tier < self::MIN_TIER_NO_UPGRADE_PAGE) {
```

Comment:
We could also display the page if the Premium key is not valid but not sure current design works for that.

Reply:
Could you please elaborate, why can't this page be shown when the Premium key is invalid?

Reply:
> why can't this page be shown when the Premium key is invalid?

It was not sure if we wanted to change existing behavior. 
I've tested the page without a plan, with an invalid key and with a starter key and looks good, so added in this [commit](https://github.com/mailpoet/mailpoet/pull/5508/commits/0c7251de607fadc9929450a1bcfb4e587c9c18c9)

Reply:
Thanks for making the change!